File: packages/core/src/xmn/client-identity.ts
/*------------------------------------------------------------------------------
   About      : Represents a Mubble Client Installation.
   
   Created on : Sat Jun 24 2017
   Author     : Raghvendra Varma
   
   Copyright (c) 2017 Mubble Networks Private Limited. All rights reserved.
------------------------------------------------------------------------------*/

import { SyncRequest } from '../master' 

/**
 * It is allowed to add members to this interface by extending it
 * Any new member must be either string or a number
 * plain objects 
 */
export interface ClientIdentity {

  uniqueId      : string    // Typically a version in 'a.b.c' format 

  // only valid when request is from Mubble (web)app
  appName       : string
  channel       : string
  appVersion    : string
  jsVersion     : string

  // only available when client is issued an identity
  clientId      : number
  userLinkId    : string
  userName      : string

  // TODO: Remove this
  firstName     : string
  lastName      : string
  // Information passed by the client used by   
  location      : string   // it is in form of serialized json object 
  networkType   : string

  syncReq      ?: SyncRequest
}

File: packages/android/java/xmn/client-identity.kt
package xmn

import core.JsonSerializable
import org.json.JSONObject

/**
 * Created by
 * siddharthgarg on 04/04/18.
 */

data class ClientIdentity(val appName: String, val channel: String,
                          val appVersion: String): JsonSerializable {

  constructor(json: JSONObject)
      : this(json.getString("appName"), json.getString("channel"), json.getString("appVersion"))

  var jsVersion       : String?     = null

  // only available when client is issued an identity
  var clientId        : Long?       = null
  var userLinkId      : String?     = null
  var userName        : String?     = null
  var firstName       : String?     = null
  var lastName        : String?     = null
  var migMobileNo     : String?     = null

  var userRole        : String?     = null
  var profilePicMd5   : String?     = null
  var settingsMd5     : String?     = null

  override fun toJsonObject(): JSONObject {

    val obj = JSONObject()

    obj.put("appName",        appName)
    obj.put("channel",        channel)
    obj.put("appVersion",     appVersion)
    obj.put("jsVersion",      jsVersion)
    obj.put("clientId",       clientId)
    obj.put("migMobileNo",    migMobileNo)
    obj.put("userLinkId",     userLinkId)
    obj.put("userName",       userName)
    obj.put("firstName",      firstName)
    obj.put("lastName",       lastName)
    obj.put("userRole",       userRole)
    obj.put("profilePicMd5",  profilePicMd5)
    obj.put("settingsMd5",    settingsMd5)

    return obj
  }

}
